pset-iterators.js

```diff
@@ -171,6 +243,33 @@ console.log('Problem 6:')
 
 // Add your code below this line
 
+const shortTermPresidents = function(){
+
+  // remove the president who hasn't left office yet
+  presidents.pop()
+
+  // create a new array of presidental objects with added integer properties
+  const presidentsWithAddedProperties = presidents.map(function(president){
+    president.took_office_year = parseInt(president.took_office.substring(0, 4))
+    president.left_office_year = parseInt(president.left_office.substring(0, 4))
+    return president
+  })
+
+  // filter out presidents who served less than four years
+  const presidentsWhoServedLessThanFourYears = presidentsWithAddedProperties.filter(function(president){
+    return president.left_office_year - president.took_office_year < 4
+  })
+
+  const namesOfPresidentsWhoServedLessThanFourYears = presidentsWhoServedLessThanFourYears.map(function(president){
```

Comment:
@carly-ga you could have left this step off, returning a filtered array of president objects would have been fine too.